beetsplug/importhistory.py

```diff
@@ -0,0 +1,157 @@
+"""Adds a `source_path` attribute to imported albums indicating from what path
+the album was imported from. Also suggests removing that source path in case
+you've removed the album from the library.
+
+"""
+
+import os
+import re
+from shutil import rmtree
+
+from beets.plugins import BeetsPlugin
+from beets.ui import colorize as colorize_text
+from beets.ui import input_options, input_yn
+
+
+class ImportHistPlugin(BeetsPlugin):
+    """Main plugin class."""
+
+    def __init__(self):
+        """Initialize the plugin and read configuration."""
+        super(ImportHistPlugin, self).__init__()
+        self.config.add(
+            {
+                "auto": True,
+            }
+        )
+        if self.config["auto"]:
+            self.import_stages = [self.import_stage]
+            self.register_listener("item_removed", self.suggest_removal)
+        # In order to stop suggestions for certain albums in
+        # self.suggest_removal, we initialize an empty list of mb_albumids that
+        # these will be ignored in future runs
+        self.stop_suggestions_for_albums = []
+
+    def import_stage(self, _, task):
+        """Event handler for albums import finished."""
+        # first, we sort the items using a dictionary and a list of paths for
+        # every mbid
+        for item in task.imported_items():
+            item["source_path"] = item.path
+            item.try_sync(write=True, move=False)
+
+    def suggest_removal(self, item):
+        """Prompts the user to delete the original path the item was imported from."""
+        if "source_path" not in item:
+            try:
+                self._log.warn(
+                    "Item without a source_path was found: {0.title} by {0.artist}",
+                    item,
+                )
+            except UnicodeDecodeError:
+                self._log.warn(
+                    "Item without a source_path was found: mb_trackid: {0.mb_trackid}",
+                    item,
+                )
+            return
+        if (
+            not item["source_path"]
+            or item.mb_albumid in self.stop_suggestions_for_albums
+        ):
+            return
+        if os.path.isdir(item["source_path"]):
+            # We ask the user whether they'd like to delete the item's source
+            # directory
+            delete = input_yn(
+                "The item:\n{path}\nis originated in the directory:\n{source}\n"
+                "Would you like to delete the source directory of this item?".format(
+                    path=colorize_text(
+                        "text_warning", item.path.decode("utf-8")
```

Comment:
Instead of `_.decode("utf-8")`, we have a utility function for cases like this, called `displayable_path`:
https://github.com/beetbox/beets/blob/0604aeadc1d4fd77d3df64b6f14d4c43a21a2294/beets/util/__init__.py#L408

Reply:
> we have a utility function for cases like this, called `displayable_path`

Thanks :) Switched to it, but haven't tested it yet.